tests/unit/plugins/inventory/test_opennebula.py

```diff
@@ -58,7 +58,21 @@ def get_vm_pool():
         'ETIME': 0,
         'GID': 132,
         'GNAME': 'CSApparelVDC',
-        'HISTORY_RECORDS': {},
+        'HISTORY_RECORDS': [
+            {
+                'HISTORY': OrderedDict({
+                    'OID': '42',
```

Comment:
This data structure looks very different from what the code is actually using. Here `HISTORY_RECORDS` is a list of dictionaries with a key `HISTORY` in them, and that in turn gives a dictionary with a key `HOSTNAME` in it.

The code expects `HISTORY_RECORDS` to be an object with an attribute `HISTORY`, which is a list, and contains objects that have an attribute called `HOSTNAME`.

Reply:
I'm afraid I'm completely lost, this is way beyond my ability. Hoping the original author @feldsam can help.